### PR TITLE
Fix for Krell openspeedshop spack package bug.  New multi-value variant for GUI build.

### PR DESCRIPTION
**Pull request by jgalarowicz**
Fix a bug where openspeedshop will not build when no mpi variants are specified.  Also switch to a multiple level variant for building the gui(s).  Use none, qt3, and qt4 as the variants with qt3 being the default.

### CODE REVIEW

**Comment by adamjstewart** [comment]
I guess I'll close #4848 then. I assume you've seen #4843?

**Comment by jgalarowicz** [comment]
Sorry I didn't see these other issues.   I will look at the libtool-ltdl-devel dependency problem.

**Comment by jgalarowicz** [comment]
@adamjstewart Is this one good to go, or do I need additional changes?   Thanks!

File: var/spack/repos/builtin/packages/openspeedshop/package.py [hunk 1]
```diff
@@ -77,14 +77,10 @@ class Openspeedshop(CMakePackage):
...
     variant('cuda', default=False,
             description="build with cuda packages included.")
-    variant('useqt4gui', default=False,
-            description="build with Qt4/Qt5 based gui package enabled. \
-                         Do not build older Qt3 gui")
-    variant('rtfe', default=False,
-            description="build for clusters heterogeneous processors \
-                         on fe/be nodes.")
-    variant('build_type', default='None', values=('None'),
-            description='CMake build type')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/openspeedshop/package.py]: You need to keep this variant. This is what controls the CMake build type now, we got rid of the `build_type` function.

**Comment by jgalarowicz** [comment]
@adamjstewart 
 Hi Adam - I updated the package file but not sure how to get it committed.

[jeg@localhost packages]$ 
[jeg@localhost packages]$ git commit -a "Add the build_type variant back into the openspeedshop package file." openspeedshop/package.py
fatal: Paths with -a does not make sense.
[jeg@localhost packages]$ git commit -m "Add the build_type variant back into the openspeedshop package file." openspeedshop/package.py
fatal: cannot do a partial commit during a merge.
[jeg@localhost packages]$ vi !$
vi openspeedshop/package.py
[jeg@localhost packages]$

**Comment by adamjstewart** [comment]
Try `git commit -am`. Or `git add && git commit -m`.

**Comment by adamjstewart** [comment]
You might also be in the middle of a merge. `git status` would tell you.

**Comment by jgalarowicz** [comment]
[jeg@localhost packages]$ git add openspeedshop/package.py
[jeg@localhost packages]$ git commit -m "Add the build_type variant back 
into the openspeedshop package file." openspeedshop/package.py
fatal: cannot do a partial commit during a merge.
[jeg@localhost packages]$

**Comment by jgalarowicz** [comment]
I guess I am.   So, I wait until that is finished (when git status says 
it is not merging)?

# On branch krell_spack_updates
# All conflicts fixed but you are still merging.
#   (use "git commit" to conclude merge)
#
# Changes to be committed:
#
#    modified:   ../../../../../lib/spack/docs/packaging_guide.rst
#    modified: ../../../../../lib/spack/spack/build_systems/autotools.py
#    modified: ../../../../../lib/spack/spack/build_systems/cmake.py
#    modified:   ../../../../../lib/spack/spack/test/url_parse.py
#    modified:   ../../../../../lib/spack/spack/test/versions.py
#    modified:   ../../../../../lib/spack/spack/url.py
#    modified:   ../../../../../lib/spack/spack/version.py
#    modified:   alquimia/package.py
#    modified:   benchmark/package.py
#    modified:   bml/package.py
#    modified:   caffe/package.py
#    modified:   callpath/package.py
#    modified:   cbtf-argonavis/package.py
#    modified:   cbtf-krell/package.py
#    modified:   cbtf-lanl/package.py
#    modified:   cbtf/package.py
#    modified:   cdo/package.py
#    modified:   clamr/package.py
#    new file:   cleaveland4/package.py
#    modified:   clhep/package.py
#    modified:   dealii/package.py
#    modified:   eccodes/package.py
#    modified:   eigen/package.py
#    modified:   elemental/package.py
#    modified:   espressopp/package.py
#    modified:   fenics/package.py
#    modified:   flecsale/package.py
#    modified:   flecsi/package.py
#    modified:   geant4/package.py
#    modified:   gmsh/package.py
#    modified:   gobject-introspection/package.py
#    modified:   gromacs/package.py
#    modified:   hpx/package.py
#    modified:   jdk/package.py
#    modified:   lbann/package.py
#    modified:   legion/package.py
#    modified:   llvm/package.py
#    modified:   lua/package.py
#    modified:   mad-numdiff/package.py
#    modified:   magics/package.py
#    modified:   nalu/package.py
#    modified:   nekbone/package.py
#    modified:   openblas/package.py
#    modified:   opencoarrays/package.py
#    modified:   openspeedshop/package.py
#    modified:   pegtl/package.py
#    new file:   perl-math-cdf/package.py
#    modified:   portage/package.py
#    modified:   qt/package.py
#    modified:   quinoa/package.py
#    modified:   relion/package.py
#    modified:   root/package.py
#    modified:   sas/package.py
#    new file:   shortstack/package.py
#    new file:   snptest/package.py
#    new file:   stacks/package.py
#    new file:   stringtie/package.py
#    new file:   structure/package.py
#    modified:   sublime-text/package.py
#    new file:   subread/package.py
#    new file:   sumaclust/package.py
#    new file:   swarm/package.py
#    modified:   symengine/package.py
#    new file:   tabix/package.py
#    modified:   tmux/package.py
#    new file:   transdecoder/package.py
#    new file:   transposome/package.py
#    modified:   trilinos/package.py
#    new file:   trimgalore/package.py
#    modified:   vc/package.py
#    modified:   vecgeom/package.py
#    new file:   viennarna/package.py
#    modified:   votca-csg/package.py
#    modified:   votca-ctp/package.py
#    modified:   votca-moo/package.py
#    modified:   votca-tools/package.py
#    modified:   votca-xtp/package.py
#    modified:   vpic/package.py
#    modified:   xsdktrilinos/package.py
#    modified:   zstd/package.py
#

**Comment by adamjstewart** [comment]
Yes, you'll need to finish the merge, or abort it, before you can add any new commits.

**Pull request merged**